bin/tests/integration/named_metrics_tests.rs

```diff
@@ -604,7 +605,75 @@ fn test_updates() {
     let server_path = Path::new(&server_path);
     let database =
         server_path.join("tests/test-data/test_configs/example.com_dnssec_update_2.jrnl");
-    std::fs::remove_file(&database).expect("failed to cleanup after test");
+    fs::remove_file(&database).expect("failed to cleanup after test");
+}
+
+#[test]
+#[cfg(all(feature = "__tls", feature = "recursor", feature = "metrics"))]
+fn test_opp_enc_metrics() {
+    subscribe();
+
+    named_test_harness("example_recursor_opportunistic_enc.toml", |socket_ports| {
```

Comment:
This test has the potential to fight with `rfc_9539_smoke_test` over the state file, since they're both using the same configuration file, and they're both in the same target. We should either split configuration files and state files (akin to `dnssec_with_update_2.toml`) or fold the smoke test into this test.